pkg/controller/workload/kubernetes/application/application.go

```diff
@@ -271,6 +286,29 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		return reconcile.Result{Requeue: false}, errors.Wrapf(err, "cannot get %s %s", v1alpha1.KubernetesApplicationKind, req.NamespacedName)
 	}
 
+	if meta.WasDeleted(app) {
```

Comment:
@muvaf since the KARs have owner references back to the KA, do we actually need to manually call delete, or can we just add the finalizer and keep checking if the KARs have been deleted here?

Reply:
Yes, because:
* background deletion: delete child resources after the parent disappears
* foreground deletion: delete child resources immediately, if all is gone then make the parent disappear

and the default is background deletion. We don't let the parent to disappear via finalizer, so, children never get deletion calls if we don't call deletion on them.